bodhi/models/models.py

```diff
@@ -1077,6 +1077,16 @@ def set_request(self, db, action, username):
                 update=self, agent=username))
             return
 
+        # If an update with pending status receives negative karma, make sure it gets obsoleted
+        if self.status is UpdateStatus.pending and self.request is UpdateRequest.testing \
+            and self.karma < 0:
```

Comment:
I'm thinking that we should only do this if autokarma is turned on. Do you agree?